package/upgrade/upgrade_node.sh

```diff
@@ -186,14 +186,34 @@ shutdown_non_migrate_able_vms()
     done
 
   # VMs with nodeAffinity
+  # Skip only when the VMs have other places to go
+  local node_labels=""
+  local node_count=0
   kubectl get vmi -A -l kubevirt.io/nodeName=$HARVESTER_UPGRADE_NODE_NAME -o json |
     jq -r '.items[] | select(.spec.affinity.nodeAffinity != null) | [.metadata.name, .metadata.namespace] | @tsv' |
     while IFS=$'\t' read -r name namespace; do
       if [ -z "$name" ]; then
         break
       fi
-      echo "Stop ${namespace}/${name}"
-      virtctl stop $name -n $namespace
+      node_labels=$(kubectl -n "$namespace" get vmi "$name" -o yaml |
+        yq '.spec.affinity.nodeAffinity.*.nodeSelectorTerms[].matchExpressions[] | select(.operator=="In" and .values[] == "true") | .key')
+      for nl in $node_labels; do
+        node_count=$(kubectl get nodes -o yaml |
+              nl="$nl" yq '.items[].metadata | select(.labels.[env(nl)]=="true") | .name' | wc -l)
```

Comment:
Do we need to check whether this node is cordoned?
```bash
kubectl get nodes -o yaml | nl="$nl" yq '.items[] | select(.metadata.labels.[env(nl)]=="true" and .spec.unschedulable==null) | .metadata.name'
```

Reply:
Thanks for the notice! I have updated it, and please check it again.